presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -558,6 +567,22 @@ int32_t SystemConfig::oldTaskCleanUpMs() const {
   return optionalProperty<int32_t>(kOldTaskCleanUpMs).value();
 }
 
+ConnectorConfig::ConnectorConfig() {
+  registeredProps_ =
+      std::unordered_map<std::string, folly::Optional<std::string>>{
+          NONE_PROP(kConnectorName),
+          BOOL_PROP(kHiveImmutablePartitions, false),
```

Comment:
Same question here. ConnectorConfig seems to be generic but is tied to a Hive property.

Reply:
@majetideepak 
We might need to revisit this at some point. Worth looking how Java deals with different props for different connector configs and copy the behavior.
cc @kewang1024